### PR TITLE
Added Detailed Email Error-Handling + Nice Email Copy

### PR DESCRIPTION
**Pull request by aivantg**
[//]: # "These comments meant for your reference, they are invisible and don't need to be deleted"

## What's new in this PR?

This PR enhances Bill Generation by:
- sending useful emails to the admin when bill generation stops or errors in both defined and undefined ways with instructions on fixing
- sending a success email to the admin where they can review the bill in the email AND click a link that will automatically approve it
- cleans up code organization a bit

I only made custom errors for the most actionable and common errors, but more could easily be added. 

[//]: # "Describe what's new in this PR"

### How to Review

You should be able to test on master! Currently emails go to aivantgoyal@gmail.com, but I can switch the email to you if you're interested, just lmk :) 

### Related PRs

Includes Changes of #6 

### Next steps

We should get final copy and word on user-facing emails: Pledge Invite Email and (potentially) New Bill Email.

### Screenshots
Sample Emails:
#### Success 
image
image

#### Generic Error
image

#### Missing Enphase Data
image

#### Stale PG&E Bill
image

#### Generic PDF Generation Error
image

[//]: # "Add screenshots!!! If you'd like"

CC: @dfangshuo

### CODE REVIEW

File: Constants.js [hunk 1]
```diff
@@ -1,4 +1,13 @@
...
+  SERVER_URL: 'https://peoplepower-node.herokuapp.com',
+  PDF_DELETE_DELAY: 20,
+
+  // Mailing
+  SENDER_NAME: 'People Power Solar',
+  EMAIL_LOGO_SRC: 'https://i.imgur.com/wD9o1iM.png',
+
+  // Pledge Invites
+  INVITE_BASE_URL: 'https://peoplepower.netlify.com/onboarding',
+  INVITE_PARAM: 'invite'
```
Comment (dfangshuo) [inline Constants.js]: Need to standardize to either `invite` or `token`

File: utils/utilityApi.js [hunk 2]
```diff
@@ -7,15 +7,13 @@ A set of low-level functions that gives us the details we need from the Utility
 import fetch from 'node-fetch';
 import moment from 'moment';
 
-const API_KEY = process.env.UTILITY_API_KEY;
-const EBCERebateString = 'Credited to (Debited from) NEM Balance';
-
-// Base URL for Utility API BILLS
+// Constants
 const billsBaseUrl = 'https://utilityapi.com/api/v2/bills';
+const EBCERebateString = 'Credited to (Debited from) NEM Balance';
```
Comment (dfangshuo) [inline utils/utilityApi.js]: is there a reason you defined these here instead of putting them in the Constants.js file?

File: utils/email.js [hunk 3]
```diff
@@ -1,58 +1,29 @@
...
 
 dotenv.config(); // Set up environment variables
 
-const SENDER_NAME = ''; // specify the name of the person sending an email, for example: "Nick Wong ⚡️🔋"
-
 const transporter = nodemailer.createTransport({
   host: 'smtp.gmail.com',
-  port: 587,
-  secure: false, // true for 465, false for other ports
+  port: 465,
```
Comment (dfangshuo) [inline utils/email.js]: what's the differencce between these 2 ports?

File: utils/email.js [hunk 4]
```diff
@@ -1,58 +1,29 @@
...
 import nodemailer from 'nodemailer';
-import { getPledgeInviteById } from '../airtable/request';
+import Constants from '../Constants';
 
 dotenv.config(); // Set up environment variables
 
-const SENDER_NAME = ''; // specify the name of the person sending an email, for example: "Nick Wong ⚡️🔋"
-
 const transporter = nodemailer.createTransport({
   host: 'smtp.gmail.com',
```
Comment (dfangshuo) [inline utils/email.js]: this one's for my understanding: what does this mean?

File: utils/billgeneration.js [hunk 5]
```diff
@@ -146,19 +173,49 @@ end date: ${endDate}`
...
+    console.log('Run into error in PDF Generation process. Reporting...');
+    sendEmail(pdfBillError(subscriber, solarProject, e.message));
+    return;
+  }
+
+  // Report Success
+  const approveLink = `${Constants.SERVER_URL}/approve?id=${newBill.id}`;
+  sendEmail(
+    billSuccess(subscriber, solarProject, newBill, approveLink, localPdfPath)
+  );
```
Comment (dfangshuo) [inline utils/billgeneration.js]: is there a reason we're using this instead of generatePdf as before?

File: Constants.js (hunk 1 above)
Comment (aivantg) [reply Constants.js]: invite!

File: utils/email.js (hunk 3 above)
Comment (aivantg) [reply utils/email.js]: one works through HTTPS vs HTTP

Comment (aivantg) [reply utils/email.js]: i think LOL

File: utils/email.js (hunk 4 above)
Comment (aivantg) [reply utils/email.js]: So basically, the `nodemailer` package uses an existing mail server to send emails. Gmail accounts can be used as that "existing mail server" so `nodemailer.createTransport` is setting up nodemailer with the gmail servers

File: utils/billgeneration.js (hunk 5 above)
Comment (aivantg) [reply utils/billgeneration.js]: generatePdf is still in there! Line 182

File: utils/utilityApi.js (hunk 2 above)
Comment (aivantg) [reply utils/utilityApi.js]: I think that our constants file would be most productive being limited to things that could be considered "Parameters" or "Settings" for the app, in the sense that they are things we might want to tweak quickly without revisiting the code. 

These two variables are constants, but they are very tied to how the calls to utility API are queried. I'm assuming that if one of these changed, we'd have to rewrite the code in here too, so I grouped them together.

honestly..just kinda personal preference. we can move them if we want

**Pull request merged**